### PR TITLE
Add fuzz tests and fix parseRetryAfter overflow

### PR DESCRIPTION
**Pull request by jeremy**
## Summary

- Fix `time.Duration` overflow in `parseRetryAfter` — large Retry-After header values (e.g. `"9227000000"`) cause negative durations. Capped at 300 seconds.
- Add Go native fuzz tests for 6 parser functions across `internal/client` and `internal/commands`, satisfying the OpenSSF Scorecard Fuzzing check.
- Includes regression corpus entry for the overflow case, caught during initial fuzzing.

## Test plan

- [x] `FuzzParseRetryAfter` regression corpus passes (previously overflowed)
- [x] All 6 fuzz test seed corpora pass as regular tests
- [x] `make build` succeeds
- [x] `make test-unit` passes

---
## Summary by cubic
Fixes `parseRetryAfter` overflow by capping numeric and HTTP-date `Retry-After` at 300s to prevent negative durations. Adds fuzz tests for six parser helpers with a regression seed, adds unit tests for the cap (meets OpenSSF Scorecard Fuzzing), and hoists the `parseLinkNext` regexp to a package var.

<sup>Written for commit 265537b523509d88b9aaab203c2e7e245e41a835. Summary will update on new commits.</sup>

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -309,6 +309,10 @@ func parseRetryAfter(value string) time.Duration {
 		return time.Second
 	}
 	if seconds, err := strconv.Atoi(value); err == nil && seconds > 0 {
+		const maxRetryDelay = 300 // 5 minutes; anything larger is almost certainly bogus
+		if seconds > maxRetryDelay {
+			seconds = maxRetryDelay
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/client/client.go]: Intentional — the cap prevents `time.Duration` overflow which produces *negative* durations (the bug this PR fixes). Without the cap, `9227000000 * time.Second` wraps to `-2561040h`. A legitimate server asking for >5min retry is vanishingly unlikely from our API.

File: internal/client/client.go [hunk 2]
```diff
@@ -309,6 +309,10 @@ func parseRetryAfter(value string) time.Duration {
...
+		}
 		return time.Duration(seconds) * time.Second
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/client/client.go]: Fixed — added `capped` (600→300s), `overflow` (9227000000→300s), `zero`, and `negative` test cases to `TestParseRetryAfter`.

File: internal/client/fuzz_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	f.Add("")
+	f.Add(`<https://api.example.com/page2>; rel="next"`)
+	f.Add(`<https://api.example.com/page1>; rel="prev", <https://api.example.com/page3>; rel="next"`)
+	f.Add(`<https://api.example.com/page1>; rel="prev"`)
+	f.Add(`<>; rel="next"`)
+	f.Add(`malformed header`)
+
+	f.Fuzz(func(t *testing.T, header string) {
+		parseLinkNext(header) // must not panic
+	})
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/client/fuzz_test.go:15]: Fixed — hoisted regexp to package-level `var linkNextRe`.

File: internal/client/client.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/client/client.go]: Fixed — HTTP-date branch now also caps at `maxRetryDelay` (300s). Hoisted the const to function scope so both branches share it.
